src/test/java/org/opendevstack/provision/authentication/crowd/ProvAppSimpleCachingGroupMembershipManagerTest.java

```diff
@@ -41,7 +42,7 @@
 
   @MockBean private GroupManager groupManager;
 
-  @MockBean private BasicCache cache;
+  @Mock private BasicCache cache;
```

Comment:
`@MockBean private BasisCache cache` looked ok for me... why you change it to use mockito `@Mock` annotation... 
the idea is to get rid of mockito annotation in test classes because it causes sometime confusion and troubles when mixing it with spring mockito support in same test class.

Reply:
Hm yes I can't really remember I have to admit, but looking at the difference of ´@Mock´, ´@InjectMocks´ and ´@MockBean´ I have to say that I don't agree that we should only use ´@MockBean´. They have differences and can be potentially used for different scenarios.
One major difference for example is that `@Mock` just creates simple Mock objects which are way more performant in the test execution than ´@MockBean´. ´@MockBean´ can be used to add a Mock into the Spring application context (or replace an existing bean in there).
So I'd say, whenever a Mock does not need to be in the application context, it makes sense considering ´@Mock´ instead of ´@MockBean´. ´@InjectMocks´ creates an actual object but allows to use/inject mocks for dependencies of this class. I will need to get myself more familiar with the different scenarios and when which to use. Thus I'd rather wait with this PR. If you have any further insight please lets discuss.

Reply:
ok after reading through https://rieckpil.de/difference-between-mock-and-mockbean-spring-boot-applications/ I think we can improve very much the test situation and performance compared to its current state. 

As a rule of thumb I'd suggest the following approach (not sure if it works everywhere yet though)
- Controller test classes get the spring boot context by the SpringBootTest and MockBean annotations
- Service test classes only have ExtendWith(MockitoExtension.class) annotation and the service the test classes tests gets InjectMocks annotation while its dependencies (collaborateurs) are annotated with Mock if necessary.

I'll try this out with the codebase of this PR.

Reply:
I'd like to stick first to `@MockBean` and improve it later.
To avoid to delay much the review and merge of this PR, I'll propose to do the "situation and performance improvement in a 2nd change after finalizing and merging this one.
Sounds good?

Reply:
yes ok, I already changed it for a handful smaller tests class were it was applicable and easy to fix (see latest commit: https://github.com/opendevstack/ods-provisioning-app/pull/659/commits/e46af5e24b447f75a13dab40e03264dc31b152e3). I think I'll stop it here then until we merged.